src/main/scala/io/iohk/ethereum/db/storage/NodeStorage.scala

```diff
@@ -9,12 +9,33 @@ import io.iohk.ethereum.db.storage.NodeStorage._
   *   Key: hash of the RLP encoded node
   *   Value: the RLP encoded node
   */
-class NodeStorage(val dataSource: DataSource) extends KeyValueStorage[NodeHash, NodeEncoded, NodeStorage] {
+class NodeStorage(val dataSource: DataSource) {
 
   val namespace: IndexedSeq[Byte] = Namespaces.NodeNamespace
-  def keySerializer: NodeHash => IndexedSeq[Byte] = _.toIndexedSeq
-  def valueSerializer: NodeEncoded => IndexedSeq[Byte] = _.toIndexedSeq
-  def valueDeserializer: IndexedSeq[Byte] => NodeEncoded = _.toArray
+  private val specialNameSpace = namespace.head
+
+  def specialSerializer(nodeHash: NodeHash): Array[Byte] = {
```

Comment:
I'm not sure if I understand the naming here. Why is it _special_ ?

Reply:
hmm naming probably could be better :) but it is special in that sense that it differs from all our aother serializes which converts `K` to `IndexedSeq[Byte]`. This one just prepend namespace byte to bytestring and gives us Array[Byte], shortcuting this our normal serialization `K -> IndexedSeq[Byte]-> Array[Byte]`